common/testData/localeIdentifiers/localeCanonicalization.txt

```diff
@@ -0,0 +1,1647 @@
+# Test data for locale identifier canonicalization
+#  Copyright © 1991-2020 Unicode, Inc.
+#  For terms of use, see http://www.unicode.org/copyright.html
+#  Unicode and the Unicode Logo are registered trademarks of Unicode, Inc. in the U.S. and other countries.
+#  CLDR data files are interpreted according to the LDML specification (http://unicode.org/reports/tr35/)
+#
+# Format:
+# <source locale identifier>	;	<expected canonicalized locale identifier>
+#
+# The data lines are divided into 4 sets:
+#   explicit: a short list of explicit test cases.
+#   fromAliases: test cases generated from the alias data.
+#   decanonicalized: test cases generated by reversing the normalization process.
+#   withIrrelevants: test cases generated from the others by adding irrelevant fields where possible,
+# to ensure that the canonicalization implementation is not sensitive to irrelevant fields. These include:
+#     Language: aaa
+#     Script:   Adlm
+#     Region:   AC
+#     Variant:  fonipa
+######
+
+
+
+# explicit
+
+art_lojban	;	jbo
+en_US_aaland	;	en_US
+en_US_heploc	;	en_US_alalc97
+en_US_polytoni	;	en_US_polyton
+en_aaland	;	en_AX
+en_arevela	;	en
+en_arevmda_arevela	;	en
+en_lojban	;	en
+hy_arevela	;	hy
+hy_arevmda	;	hyw
+hy_arevmda_arevela	;	hyw
+hye_arevmda	;	hyw
+no_bokmal_nynorsk	;	nb
+no_nynorsk_bokmal	;	nb
```

Comment:
This imply, in the algorithm, we need to sort the variant first before look up , so we will first search for no_bokmal before searching no_nynorsk, right?

Reply:
Yes, that's in the notes. You have to sort first by the number of fields then alphabetically. Of course, the implementation can use a much faster algorithm than CLDRs with a speedier lookup; this is to be a simple reference implementation.